packages/consensus/src/validation/block.cairo

```diff
@@ -47,6 +50,50 @@ pub fn compute_and_validate_tx_data(
 
         let txid = double_sha256_byte_array(tx_bytes_legacy);
 
+        // Removes the outpoint hash of a transaction input if it was in the cache.
+        let inputs = *tx.inputs;
+        let mut j = 0;
+        while j != inputs.len() {
+            let outpoint = (*inputs[j]).previous_output;
+            let outpoint_hash = PoseidonTrait::new().update_with(outpoint).finalize();
```

Comment:
The hashing and updating the cache should be encapsulated in utxoset

Reply:
what do you mean?